com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/SyncFlowableStepWithHooks.java

```diff
@@ -89,8 +97,17 @@ private Module determineModuleFromAppName(DeploymentDescriptor deploymentDescrip
     }
 
     protected String getModuleName(CloudApplicationExtended cloudApplication) {
-        return ApplicationMtaMetadataParser.parseAppMetadata(cloudApplication)
-                                           .getModuleName();
+        return getApplicationMtaMetadata(cloudApplication)
+            .getDeployedMtaModule()
+            .getModuleName();
+    }
+
+    private ApplicationMtaMetadata getApplicationMtaMetadata(CloudApplication app) {
```

Comment:
This method is duplicated a lot in many files.
Maybe introduce a Factory for the metadata?